lib/worker/syncer.ex

```diff
@@ -151,11 +151,25 @@ defmodule BorsNG.Worker.Syncer do
   @spec sync_user(GitHub.User.t) :: %User{}
   def sync_user(gh_user) do
     case Repo.get_by(User, user_xref: gh_user.id) do
-      nil -> Repo.insert!(%User{
-        user_xref: gh_user.id,
-        login: gh_user.login})
+      nil ->
+        case Repo.get_by(User, login: gh_user.login) do
+          nil ->
+            Repo.insert!(%User{
+              user_xref: gh_user.id,
+              login: gh_user.login})
+          user ->
+            if user.user_xref != gh_user.id do
+              Logger.debug("Syncer: sync_user: github user #{inspect(gh_user.login)} changed id from #{inspect(user.user_xref)} to #{inspect(gh_user.id)}")
+              user
+              |> User.changeset(%{user_xref: gh_user.id})
```

Comment:
That is not correct. How about, instead, making it

```suggestion
              |> User.changeset(%{login: "#{user.login}/renamed/#{user.id}"})
```

and then performing the insert as normal afterward. This way, claiming a username after the original user goes away doesn't [implicitly give you the old user's perms](https://github.com/rust-lang/crates.io/issues/1584). Since github usernames aren't allowed to have slashes in them, renaming it like this should ensure there's no conflicts. It's called `{name}/rename/{id}` since github logins never have slashes in them, and the ID numbers themselves are unique, so there's no possibility of conflict.